test/test_other.py

```diff
@@ -380,7 +380,15 @@ def test_emcc_output_worker_mjs(self, args):
                       test_file('hello_world.c')] + args)
     src = read_file('subdir/hello_world.mjs')
     self.assertContained("new URL('hello_world.wasm', import.meta.url)", src)
-    self.assertContained("new Worker(new URL('hello_world.mjs', import.meta.url), workerOptions)", src)
+    self.assertContained("""new Worker(new URL('hello_world.mjs', import.meta.url), {
+        'type': 'module',
+        // This is the way that we signal to the node worker that it is hosting
+        // a pthread.
+        'workerData': 'em-pthread',
+        // This is the way that we signal to the Web Worker that it is hosting
+        // a pthread.
+        'name': 'em-pthread',
+    })""", src)
```

Comment:
Rather than checking for this specific output I wonder if we can instead test directly that it works with vite.

I recently added some vite tests to test_browser.py so I think we could add them there.   Then we could test that it actually runs though vite and executed.

I also wonder if its enough to test this in just rollup, since IIUC vite uses rollup?

Reply:
Awesome! I'll look into adding some tests there 👀 

For this particular Vite issue, rollup would not be enough to test. The special logic around these workers is implemented on the Vite side.

Reply:
Good to know.  Thanks for working on this!

As a followup question.  Is there some way we can test this stuff using node, or do we really need to test the output in a browser?  I test to prefer non-browser-tests since they are much easy to work with, but maybe this is a case where we really need the browser?

Reply:
For the particular issue concerning the PR, I think the browser is the only way.

Vite is primarily a bundler for the browser, although, as it is built on top of rollup it can be customised and you could bundle a JS file only and test with node (providing the bundled code is Node compatible).

I think the [documentation for bundling a library](https://vite.dev/guide/build#library-mode) could possibly be used by some tests and then tested on the command line with node. I could try create PoC, if you like?

Reply:
That would be most welcome.

BTW, are you a vite contributor/maintainer?

Reply:
> are you a vite contributor/maintainer?

No, I'm not. I have contributed a couple of small updates to Vite to fix issues with Emscripten. The Vite team has been very responsive to helping solve them (https://github.com/vitejs/vite/pull/14113, https://github.com/vitejs/vite/pull/19010).